libfwupd/fwupd-remote.c

```diff
@@ -320,15 +332,19 @@ fwupd_remote_load_from_filename (FwupdRemote *self,
 		fwupd_remote_set_filename_cache (self, filename_cache);
 	}
 
-	/* all LOCAL remotes have to include a valid File */
+	/* all LOCAL remotes have to include a valid MetadataURI */
 	if (self->kind == FWUPD_REMOTE_KIND_LOCAL) {
-		g_autofree gchar *filename_cache = NULL;
-		filename_cache = g_key_file_get_string (kf, group, "File", error);
-		if (filename_cache == NULL)
-			return FALSE;
+		const gchar *filename_cache = metadata_uri;
+		if (g_str_has_prefix (filename_cache, "file://"))
+			filename_cache += 7;
```

Comment:
To my point above, if you just looked for the prefix of file:// you could handle this whole section without needing to look for the kind.